jobs/involuntary-dissolutions/involuntary_dissolutions.py

```diff
@@ -64,12 +69,59 @@ def shell_context():
 
     app.shell_context_processor(shell_context)
 
+def initiate_dissolution_process(app: Flask):  # pylint: disable=redefined-outer-name
+    """Initiate dissolution process for new businesses where AR has not been filed for 2 yrs and 2 months."""
+    try:
+        # check if batch has already run today
+        batch_today = Batch.find_by(batch_type=Batch.BatchType.INVOLUNTARY_DISSOLUTION).filter(Batch.start_date == datetime.today())
+        if batch_today:
+            app.logger.debug('Skipping job run since batch job has already run today.')
+            return
+        
+        # check if batch can be run today
+        new_dissolutions_schedule_config = Configuration.find_by_name(config_name='DISSOLUTIONS_STAGE_1_SCHEDULE')
+        tz = pytz.timezone('US/Pacific')
+        cron_valid = croniter.match(new_dissolutions_schedule_config.val, tz.localize(datetime.today()))
+        if not cron_valid:
+            app.logger.debug('Skipping job run since current day of the week does not match the cron schedule.')
+            return
```

Comment:
I think it would be better to move this check out of initiate dissolution process(stage 1) because there will also be stage 2 and stage 3 checks to see if they need to run today. They can be combined together. 🤔

Reply:
Good idea! Do you think it also makes sense to move the "only run once a day" check as well? Or will the other stages run multiple times a day?

Reply:
🤔 I don't think other stages will need this check as per the tech design and my conversation with Argus... 💭

Reply:
Sounds good, I'll only move the cron check then